compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -734,11 +734,16 @@ object Checking {
       case _ =>
         report.error(ValueClassesMayNotContainInitalization(clazz), stat.srcPos)
     }
-    if (clazz.isDerivedValueClass) {
+    // We don't check synthesised enum anonymous classes that are generated from
+    // enum extending a value class type (AnyVal or an alias of it)
+    // The error message with ID 205 will take care of generating the error message (See #22235)
```

Comment:
Write the message ID *name* instead of the numeric value? As a reader here I don't want to have to go look up what it is.